### DISCUSSION TITLE
Preparation for Upgrade 2.3.130 -> 2.3.140 // Elastic assistant

### DISCUSSION THREAD

**Discussion by mnasec** [Security-Onion-Solutions/securityonion#8444] [Unsupported Versions] [answered]
13 node grid (manager, 2 receiver, 8 search, 2 sensor).
Before upgrading from 2.3.130 to 2.3.140 (upgraded step by step from initial 2.3.30) i ran the upgrade assistant for elastic. It found the following criticals and warnings:
**Elasticsearch:**
```
Critical| Date fields use deprecated camel case formats | Index | elastalert_status_status | Manual
Critical | Date fields use deprecated camel case formats | Index | elastalert_status | Manual
Critical | Date fields use deprecated camel case formats | Index | elastalert_status_past | Manual
Critical | Date fields use deprecated camel case formats | Index | elastalert_status_silence | Manual
Critical | Date fields use deprecated camel case formats | Index | elastalert_status_error | Manual
Warning | Disk watermarks do not treat single-node clusters differently in versions 8.0 and later. | Node |   | Manual
```

From the docu i found:
_Upgrade Assistant may display issues for [ElastAlert](https://docs.securityonion.net/en/2.3/elastalert.html#elastalert) indices, but they will be deleted automatically when you update. You may need to investigate any non-Elastalert issues._

Question: Should i change anything before upgrading?

**Kibana:**
```
Critical | Setting "xpack.ml.enabled" is deprecated | Config | Manual
Critical | Setting "cpu.cgroup.path.override" is deprecated | Config | Manual
Critical | Setting "cpuacct.cgroup.path.override" is deprecated | Config | Manual
Warning | Setting "server.rewriteBasePath" should be set when using "server.basePath" | Config | Manual
Warning | Kibana's logging format is changing in 8.0 | Config | Manual
Warning | The "xpack.reporting.roles" setting is deprecated | Config | Manual
Warning | Setting "security.showInsecureClusterWarning" is deprecated | Config | Manual
Warning | "xpack.security.session.idleTimeout" is now 8 hours | Config | Manual
Warning | "xpack.security.session.lifespan" is now 30 days | Config | Manual
Warning | Found index patterns using scripted fields | Uncategorized | Manual
Warning | The Security feature is changing, and the "auditor" role requires an update | Feature |  
```

Question: Here the same - should i change anything before upgrading or does soup cover these criticals and warnings?

**Comment by weslambert** [comment, marked as answer]
I think you should be okay. Even for the `CRITICAL` items, I don't think those are items that would prevent an upgrade from completing successfully. The configuration should be updated appropriately during the upgrade. I would ensure you are checking against each of the items referenced in the upgrade notes, otherwise.

https://docs.securityonion.net/en/latest/soup.html#elastic-8

**Reply by mnasec** [reply]
Upgrade run smoothly, thanks for the advice.
I also checked files in /opt/so/saltstack/local/salt but there was no problem.

After the upgrade on 5 nodes the minions did not run after 40 Minutes, test.ping from master did not work. I started the minions with so-salt-start and now all minions are available.

Thanks a lot!
